devenv/src/main.rs

```diff
@@ -288,50 +292,48 @@ fn main() -> Result<()> {
         log::Level::Info
     };
 
+    let logger = log::Logger::new(level);
+
+    let mut config = config::Config::load()?;
+
+    for input in cli.override_input.chunks_exact(2) {
+        config.add_input(&input[0].clone(), &input[1].clone(), &[]);
+    }
+
     let xdg_dirs = xdg::BaseDirectories::with_prefix("devenv").unwrap();
-    xdg_dirs
-        .create_data_directory(Path::new("devenv"))
-        .expect("Failed to create DEVENV_HOME directory");
     let devenv_home = xdg_dirs.get_data_home();
     let devenv_home_gc = devenv_home.join("gc");
-    std::fs::create_dir_all(&devenv_home_gc).expect("Failed to create DEVENV_HOME_GC directory");
-    let devenv_root = std::env::current_dir().expect("Failed to get current directory");
-    let devenv_dot_gc = devenv_root.join(".devenv").join("gc");
-    std::fs::create_dir_all(&devenv_dot_gc).expect("Failed to create .devenv/gc directory");
-    let devenv_dotfile = devenv_root.join(".devenv");
+
     let devenv_tmp = std::env::var("XDG_RUNTIME_DIR")
         .unwrap_or_else(|_| std::env::var("TMPDIR").unwrap_or_else(|_| "/tmp".to_string()));
-    // first 7 chars of sha256 hash of devenv_state
-    let devenv_state_hash = {
-        let mut hasher = sha2::Sha256::new();
-        hasher.update(devenv_dotfile.to_string_lossy().as_bytes());
-        let result = hasher.finalize();
-        hex::encode(result)
-    };
-    let devenv_runtime = Path::new(&devenv_tmp).join(format!("devenv-{}", &devenv_state_hash[..7]));
+
+    // using dummy values for initialization
+    let devenv_root: PathBuf = "./".into();
+    let devenv_runtime = Path::new(&devenv_tmp).join(format!("devenv-{}", 1234.to_string()));
```

Comment:
Why do we need to fake these? Surely we can fetch the current directory in most situations?

As an aside, `PathBuf::from` is probably the more rusty way of initialising a new pathbuf.